flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -1888,6 +1919,37 @@ public void requestFullScreenIntentPermission(@NonNull PermissionRequestListener
     }
   }
 
+  @RequiresApi(api = VERSION_CODES.M)
```

Comment:
The `@RequiresApi` only functions as an annotation to help warn those calling the method, which would be the plugin itself. What the method would need is a version check. Currently when either `requestNotificationPolicyAccess()` or `hasNotificationPolicyAccess()` is called on a lower Android version, a runtime exception would occur. To get around this I would suggest to

- update the code within both methods to check the Android version so that if they're on a lower Android version that a `false` boolean result will make it over to the Dart/Flutter side through either  `this.callback.complete(false)` or `result.success(false)`
- remove the `@RequiresApi` annotation on both methods as they're effectively redundant